HLTrigger/btau/plugins/HLTSumJetTag.cc

```diff
@@ -0,0 +1,149 @@
+#include "HLTSumJetTag.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "DataFormats/JetReco/interface/JetCollection.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+
+#include <numeric>
+
+template <typename T>
+HLTSumJetTag<T>::HLTSumJetTag(const edm::ParameterSet& config)
+    : HLTFilter(config),
+      m_Jets(config.getParameter<edm::InputTag>("Jets")),
+      m_JetTags(config.getParameter<edm::InputTag>("JetTags")),
+      m_JetsToken(consumes<std::vector<T>>(m_Jets)),
+      m_JetTagsToken(consumes<reco::JetTagCollection>(m_JetTags)),
+      m_MinTag(config.getParameter<double>("MinTag")),
+      m_MaxTag(config.getParameter<double>("MaxTag")),
+      m_MinJetToSum(config.getParameter<unsigned int>("MinJetToSum")),
+      m_MaxJetToSum(config.getParameter<unsigned int>("MaxJetToSum")),
```

Comment:
I think here the convention can be used as follows: if MinJetToSum is set to 0 --> no selection applied, all jets up to MaxJetToSum are stored in the output filterproduct. The default value for MinJetToSum is 1 since you want to apply a selection at least considering one jet in the event i.e. if MinJetToSum = 1 the cut is a simple selection on the leading tag-score jet in the event. Therefore, I left these quantities as unsigned int and if MinJetToSum == 0 the filter returns true right at the beginning without wasting additional CPU time. All other comments are implemented and changes will be pushed now.

Reply:
Thanks, @rgerosa . I get your point, but I find this convention potentially confusing. The filter condition (as i read it below in `::filter`) is "N_JetTags >= m_MinJetToSum *and* minTag<=Tag<=maxTag", so I would naively expect that even if one uses `m_MinJetToSum==0` the filter would return `minTag<=Tag<=maxTag` (so, not always true). So, I would suggest to avoid this convention, keeping `unsigned int` since you have a preference for that. Does that work?